### PR TITLE
Game statistics 2

### PR DESCRIPTION
**Pull request by SageOfCode**
This request will get all of my tests to pass. 

I am still working on refactoring my average goals by season method.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -70,16 +70,52 @@ def games_by_season
   end
 
   def average_goals_per_game
-    total_number_of_goals / total_number_of_games
+    (total_goals.to_f / total_number_of_games).round(2)
   end
 
-  def total_number_of_goals
-    
+  def total_goals
```
Comment (smcunning) [inline lib/stat_tracker.rb:84]: I like the name change!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -70,16 +70,52 @@ def games_by_season
...
   def total_number_of_games
+    game_count = 0
+    games.each do |game|
+      game_count += 1
+    end
+    return game_count
+  end
 
+  def average_goals_by_season
+    season_hash = { }
```
Comment (smcunning) [inline lib/stat_tracker.rb:102]: When refactoring, let's change this to a new name for clarity

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -70,16 +70,52 @@ def games_by_season
...
+    games.each do |game|
+      season_hash[game['season']] = 0
+    end
+
+    season_hash.each do |season, average_goals|
```
Comment (Stearnzy) [inline lib/stat_tracker.rb:107]: Like you mentioned, refactoring this to helper methods would be a good move here.

**Review by Stearnzy — COMMENTED** [review]
Refactor

**Review by smcunning — APPROVED** [review]
Good work!

**Pull request merged**